### ISSUE TITLE
Fails to filter fields on parent that occur after child

### ISSUE THREAD

**Issue by rhoegg** [JohnWeidner/rhChallenge#2]
Fields on parent objects that are encountered after child objects are not filtered.  Example:
```
{
    "one": 1,
    "two": 2,
    "three": {
        "one": 0.1,
        "two": 0.2,
        "three": 0.3,
        "four": 0.4
    },
    "four": 4
}
```

**Comment by JohnWeidner** [comment]
Yep.   I did not think of that situation.   Based on the requirements I
would say the correct response should be

{
    "one": 1,
    "two": 2,
    "three": {
        "one": 0.1,
        "two": 0.2,
        "three": 0.3,
    },
    "four": 4
}


Do you agree?

**Comment by JohnWeidner** [comment]
Actually

{
    "one": 1,
    "two": 2,
    "three": {
    },
    "four": 4
}

**Comment by rhoegg** [comment]
I'd say "three" and "four" should both be removed from the object with key
"three" in the parent object.  So:
{
"one": 1,
"two": 2,
"three": { },
"four": 4
}
